File: library/src/main/java/org/coner/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairWriter.java
package org.coner.crispy_fish.datatype.underscore_pairs;

import com.google.common.base.Preconditions;
import com.sun.istack.internal.NotNull;
import com.sun.istack.internal.Nullable;

import static org.coner.crispy_fish.datatype.underscore_pairs.Constants.*;

public class SimpleStringUnderscorePairWriter implements UnderscorePairWriter<String> {

    public String pair(@NotNull String key, @Nullable String value) {
        Preconditions.checkNotNull(key, "key must not be null");

        if (value == null) {
            value = "";
        }

        return UNDERSCORE + key + UNDERSCORE + value;
    }
}

File: library/src/main/kotlin/org/coner/crispy_fish/datatype/underscore_pairs/SimpleStringUnderscorePairWriter.kt
package org.coner.crispy_fish.datatype.underscore_pairs

class SimpleStringUnderscorePairWriter : UnderscorePairWriter<String> {

    override fun pair(key: String, value: String?): String {
        val writtenValue = value ?: ""
        return "_${key}_$writtenValue"
    }
}
